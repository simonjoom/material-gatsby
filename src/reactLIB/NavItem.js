import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { Link } from 'gatsby';

const NavItem = ({
  divider,
  children,
  href = '',
  onClick,
  waves,
  external,
  ...props
}) => {
  const mywaves = waves;
  const classes = cx({
    'card-image': true,
    'waves-effect': waves,
    'waves-block': waves,
    [mywaves]: waves
  });

  if (divider) return <li className="divider" />;
  const el = external ? (
    <a href={href} className={classes}>
      {children}
    </a>
  ) : (
    <Link to={href} onClick={onClick} className={classes}>
      {children}
    </Link>
  );
  return <li {...props}>{el}</li>;
};

NavItem.propTypes = {
  /**
   * children can be a string value or a node
   */
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  divider: PropTypes.bool,
  href: PropTypes.string,
  /**
   * NavItem can have onClick. If it does have, href
   * will not be rendered
   */
  onClick: PropTypes.func
};

export default NavItem;
