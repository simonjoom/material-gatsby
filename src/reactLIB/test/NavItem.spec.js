import React from 'react';
import { shallow } from 'enzyme';
import NavItem from '../src/NavItem';

describe('<NavItem />', () => {
  let wrapper;

  test('renders', () => {
    wrapper = shallow(<NavItem href="get-started.html">Getting</NavItem>);

    expect(wrapper).toMatchSnapshot();
  });

  test('passes className to first node', () => {
    wrapper = shallow(
      <NavItem className="red" href="get-started.html">
        Getting
      </NavItem>
    );

    expect(wrapper.find('li.red')).toHaveLength(1);
  });

  test('passes node as a child', () => {
    wrapper = shallow(
      <NavItem href="get-started.html">
        <p>Go Home</p>
      </NavItem>
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('can be a divider', () => {
    wrapper = shallow(<NavItem divider />);

    expect(wrapper.find('li.divider')).toHaveLength(1);
  });

  test('handles multiple childs', () => {
    wrapper = shallow(
      <NavItem href="get-started.html">
        <span>hello</span>
        <h2>World</h2>
      </NavItem>
    );

    expect(wrapper.find('span')).toHaveLength(1);
    expect(wrapper.find('h2')).toHaveLength(1);
  });

  test('passes onClick as parameter', () => {
    const onClick = jest.fn();
    wrapper = shallow(
      <NavItem onClick={onClick} href="get-started.html">
        Getting
      </NavItem>
    );

    expect(wrapper).toMatchSnapshot();

    wrapper
      .children()
      .at(0)
      .simulate('click');

    expect(onClick).toHaveBeenCalled();

    expect(
      wrapper
        .children()
        .at(0)
        .prop('href')
    ).toBeUndefined();
  });

  test('passes href as parameter', () => {
    wrapper = shallow(<NavItem href="get-started.html">Getting</NavItem>);

    expect(wrapper).toMatchSnapshot();
  });
});
