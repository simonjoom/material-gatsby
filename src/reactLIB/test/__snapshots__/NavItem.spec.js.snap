// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem /> passes href as parameter 1`] = `
<li>
  <a
    href="get-started.html"
  >
    Getting
  </a>
</li>
`;

exports[`<NavItem /> passes node as a child 1`] = `
<li>
  <a
    href="get-started.html"
  >
    <p>
      Go Home
    </p>
  </a>
</li>
`;

exports[`<NavItem /> passes onClick as parameter 1`] = `
<li>
  <a
    onClick={[MockFunction]}
  >
    Getting
  </a>
</li>
`;

exports[`<NavItem /> renders 1`] = `
<li>
  <a
    href="get-started.html"
  >
    Getting
  </a>
</li>
`;
